staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/equality.go

```diff
@@ -70,6 +79,36 @@ func getAvoidTimestampEqualities() conversion.Equalities {
 	return avoidTimestampEqualities
 }
 
+func equalIgnoringValueAtPath(a, b any, path []string) bool {
+	if len(path) == 0 { // found the value to ignore
+		return true
+	}
+	aMap, aOk := a.(map[string]any)
+	bMap, bOk := b.(map[string]any)
+	if !aOk || !bOk {
+		// Can't traverse into non-maps, handle normally
+		return avoidTimestampEqualities.DeepEqual(a, b)
+	}
+	if len(aMap) != len(bMap) {
+		return false
+	}
+	pathHead := path[0]
+	for k, aVal := range aMap {
+		bVal, ok := bMap[k]
+		if !ok {
```

Comment:
technically if `k == pathHead` and `!ok`, we should ignore that (makes the len check above more complicated). Can that happen for `metadata.managedFields`?

Reply:
I'm okay with that.  Managed fields should always be reachable by traversing maps (metadata will fail validation if not), so I'm happy to set this to whatever is the least surprising when reading the code.  Updated